src/utils/polygon.cpp

```diff
@@ -372,7 +371,6 @@ void PolygonRef::simplify(const coord_t smallest_line_segment_squared, const coo
         if ((height_2 <= 1 //Almost exactly colinear (barring rounding errors).
             && LinearAlg2D::getDist2FromLine(current, previous, next) <= 1) // make sure that height_2 is not small because of cancellation of positive and negative areas
             || (length2 < smallest_line_segment_squared
-                && next_length2 < smallest_line_segment_squared // Segments are small
```

Comment:
Hmm, I wonder if this is a safe solution.

This line is there because removing a segment may have small deviation but a large effect if that causes a very long line segment to shift one of its vertices. Some ASCII art to illustrate:

```
  ________________________________________________>
 /
|
```

These are 3 line segments: One vertical, one diagonal and one horizontal. The winding order is along the arrow, clockwise in this illustration. The algorithm may decide to remove the diagonal line segment, which makes the long horizontal line shift:

```
                                                     ________
_______...............---------------''''''''''''''''
|
```

Those shifts are visible in the print.

I'll perform some testing if this is actually happening before putting it back though. The effects of changes here are sometimes unpredictable.